python/paddle/fluid/tests/unittests/test_while_loop_op.py

```diff
@@ -0,0 +1,189 @@
+            init = result[1]
+            sums = result[2]
+            sums = layers.elementwise_add(x=init, y=sums)
+            i = layers.increment(i)
+            return [i, j, init, sums]
+
+        main_program = Program()
+        startup_program = Program()
+        with program_guard(main_program, startup_program):
+            i = layers.zeros(shape=[1], dtype='int64')
+            j = layers.zeros(shape=[1], dtype='int64')
+            init = layers.data(name="init", shape=[3, 3], dtype='float32')
+            sums = layers.data(name="sums", shape=[3, 3], dtype='float32')
+            loop_len1 = layers.fill_constant(shape=[1], dtype='int64', value=2)
+            loop_len2 = layers.fill_constant(shape=[1], dtype='int64', value=3)
+            ones = layers.fill_constant(shape=[3, 3], dtype='float32', value=1)
+
+            res = layers.while_loop(cond1, body1, [i, j, init, sums])
+
+            data1 = np.random.rand(3, 3).astype('float32')
+            data2 = np.zeros([3, 3]).astype('float32')
+
+        place = fluid.CPUPlace()
+        exe = fluid.Executor(place)
+        ret = exe.run(main_program,
+                      feed={'init': data1,
+                            'sums': data2},
+                      fetch_list=res)
+        for i in range(3):
+            data1 = np.add(data1, 1)
+            data2 = np.add(data1, data2)
+        for j in range(2):
+            data2 = np.add(data1, data2)
+        self.assertTrue(np.allclose(np.asarray(ret[3]), data2))
+
+
+class TestAPIWhileLoop_Error(unittest.TestCase):
+    def test_error(self):
+        def cond1(i):
+            return 1
+
+        def cond2(i):
+            return layers.increment(i)
+
+        def cond3(i):
+            return layers.less_than(i, ten)
+
+        def cond4(i):
+            return layers.less_than(i, ten_1)
+
+        def body(i):
+            return layers.increment(i)
+
+        main_program = Program()
+        startup_program = Program()
+        with program_guard(main_program, startup_program):
+            data1 = layers.fill_constant(shape=[1], dtype='int64', value=1)
+            data2 = layers.fill_constant(shape=[1], dtype='int64', value=1)
+            data3 = layers.fill_constant(shape=[2, 2], dtype='int64', value=1)
+            ten = layers.fill_constant(shape=[1], dtype='int64', value=10)
+            ten_1 = layers.fill_constant(shape=[2, 2], dtype='int64', value=10)
+
+            def dtype_error_cond():
+                out = layers.while_loop(data1, body, [data2])
+
+            self.assertRaises(TypeError, dtype_error_cond)
+
+            def dtype_error_body():
+                out = layers.while_loop(cond3, data1, [data2])
+
+            self.assertRaises(TypeError, dtype_error_body)
+
+            def dtype_error_loop_vars():
+                out = layers.while_loop(cond3, body, data1)
+
+            self.assertRaises(TypeError, dtype_error_loop_vars)
+
+            def dtype_error_cond_return_1():
```

Comment:
cond1, 2, 3, 4, dtype_error_cond_return_1, 2, 3, 4 are very bad names ...... to make your code readable by others or yourself in the future, do as what you have done like "dtype_error_loop_vars" (we may know it is loop vars where have problem and raises error). You should also replace "1, 2, 3, 4" by some hints when you write "dtype_error_cond_return_" and "cond"

Reply:
Done.